packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewerTextLayer.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { FC, useEffect, useRef, useState } from 'react';
+import cx from 'classnames';
+import PDFJSLib, { PDFPageProxy, PDFPageViewport, TextContent, TextContentItem } from 'pdfjs-dist';
+import { EventBus } from 'pdfjs-dist/lib/web/ui_utils';
+import { TextLayerBuilder } from 'pdfjs-dist/lib/web/text_layer_builder';
+
+const { RenderingCancelledException } = PDFJSLib as any;
+
+interface Props {
+  className?: string;
+
+  /**
+   * PDF page from pdfjs
+   */
+  loadedPage: PDFPageProxy | null | undefined;
+
+  /**
+   * Zoom factor, where `1` is equal to 100%
+   */
+  scale: number;
+
+  /**
+   * Callback for text layer info
+   */
+  setTextLayerInfo?: (info: PdfTextLayerInfo | null) => any;
+}
+
+export type PdfTextLayerInfo = {
+  /**
+   * PDF text content
+   */
+  textContent: TextContent & {
+    styles: { [styleName: string]: CSSStyleDeclaration };
+  };
+
+  /**
+   * Text span DOM elements rendered on the text layer
+   */
+  textDivs: HTMLElement[];
+
+  /**
+   * Pdf page viewport used to render text items
+   */
+  viewport: PDFPageViewport;
+
+  /**
+   * Page number, starting at 1
+   */
+  page: number;
+};
+
+type PdfTextContentInfo = {
+  /** extracted PDF text content */
+  textContent: TextContent;
+
+  /** @see Props['scale'] */
+  scale: number;
+
+  /** @see PdfTextLayerInfo['viewport'] */
+  viewport: PDFPageViewport;
+
+  /** @see PdfTextLayerInfo['page'] */
+  page: number;
```

Comment:
Same with these. If these are the same as from other types, then use `Pick<>`.

Reply:
this type is removed after refactoring :)